ModelCatalogueCorePlugin/src/groovy/org/modelcatalogue/core/export/inventory/DataModelToXlsxExporter.groovy

```diff
@@ -0,0 +1,179 @@
+package org.modelcatalogue.core.export.inventory
+
+import groovy.util.logging.Log4j
+import org.modelcatalogue.builder.spreadsheet.api.Sheet
+import org.modelcatalogue.builder.spreadsheet.api.Workbook
+import org.modelcatalogue.builder.spreadsheet.poi.PoiSpreadsheetBuilder
+import org.modelcatalogue.core.DataClass
+import org.modelcatalogue.core.DataClassService
+import org.modelcatalogue.core.DataModel
+import org.modelcatalogue.core.util.DataModelFilter
+import org.modelcatalogue.core.util.lists.ListWithTotalAndType
+
+@Log4j
+class DataModelToXlsxExporter {
+
+    DataClassService dataClassService
+    DataModel dataModel
+
+    Map<Long, DataClass> processedDataClasses = [:]
+
+    void export(OutputStream outputStream) {
+        log.info "Exporting Data Model ${dataModel.name} (${dataModel.combinedVersion}) to inventory spreadsheet."
+
+        def dataClasses = dataClassService.getTopLevelDataClasses(DataModelFilter.includes(dataModel))
+        def builder = new PoiSpreadsheetBuilder()
+        builder.build(outputStream) { Workbook workbook ->
+            apply ModelCatalogueStyles
+            sheet("DataModel") { Sheet sheet ->
+                buildOutline(sheet, dataModel, dataClasses)
+            }
+
+            // build separate sheets for each data class covered by outline
+            processedDataClasses.eachWithIndex { item, i ->
+                log.info "[${i+1}/${processedDataClasses.size()}] exporting details for data class " +
+                        "${item.value.name} (${item.value.combinedVersion})"
+                DataClassToXlsxExporter.buildDataClassDetailSheet(workbook, processedDataClasses, item.value)
+            }
+        }
+
+        log.info "data model ${dataModel.name} (${dataModel.combinedVersion}) exported to inventory spreadsheet."
+    }
+
+    private void buildOutline(Sheet sheet, DataModel dataModel, ListWithTotalAndType<DataClass> dataClasses) {
```

Comment:
passing the whole `ListWithTotalAndType` seems unnecessary. Plain `List` should be enough and you won't need to convert the list again to `ListWithTotalAndType` again later on.